src/inmanta/user_setup.py

```diff
@@ -0,0 +1,155 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import asyncio
+import secrets
+import socket
+
+import asyncpg
+import click
+
+import nacl.pwhash
+from inmanta import config, data
+from inmanta.server import config as server_config
+
+
+def generate_signing_config() -> str:
+    hostname = socket.gethostname()
+    return f"""[auth_jwt_default]
+algorithm=HS256
+sign=true
+client_types=agent,compiler,api
+key={secrets.token_urlsafe(32)}
+expire=0
+issuer=https://{hostname}:{server_config.server_bind_port.get()}/
+audience=https://{hostname}:{server_config.server_bind_port.get()}/
+"""
+
+
+def validate_server_setup() -> None:
+    """Validate the server configuration so that authentication is setup correctly."""
+    config.Config.load_config()
+
+    # make sure auth is on
+    if not server_config.server_enable_auth.get():
+        raise click.ClickException(
+            "Server authentication should be enabled before running the initial user setup. "
+            "The option auth in the server section should be enabled."
+        )
+
+    click.echo(f"Server authentication:                                  {click.style('enabled', fg='green')}")
+
+    # make sure the method is set to database
+    if server_config.server_auth_method.get() != "database":
+        raise click.ClickException(
+            "The server authentication method should be set to database to continue. Make sure auth_method in the server "
+            "section is set to database"
+        )
+
+    click.echo(f"Server authentication method:                          {click.style('database', fg='green')}")
```

Comment:
If your goal here is to create two columns that align for both output lines, Python's built-in string formatting has a functionality that might help: `f"{'hello' : <20}world"` left-aligns "hello" in a space-padded substring of 20 characters, followed by "world". So `hello               world`.

This allows you to create aligned columns without having to count spaces: `print(f"{'hello' : <20}world"); print(f"{'hello there' : <20}world")` gives
```
hello               world
hello there         world
```